DQMOffline/Trigger/plugins/JetMonitor.cc

```diff
@@ -0,0 +1,594 @@
+  std::string hist_obtag = "";
+  std::string histtitle_obtag = "";
+  std::string currentFolder = folderName_ ;
+  ibooker.setCurrentFolder(currentFolder.c_str());
+
+  if (isPFJetTrig) {hist_obtag = "pfjet";          histtitle_obtag =  "PFJet";}
+  else if (isCaloJetTrig) {hist_obtag = "calojet"; histtitle_obtag =  "CaloJet"; }
+  else {hist_obtag = "pfjet"; histtitle_obtag =  "PFJet"; } //default is pfjet 
+
+  // Test bookMESub
+  bookMESub(ibooker,a_ME,sizeof(a_ME)/sizeof(a_ME[0]),hist_obtag,histtitle_obtag,"","");
+  bookMESub(ibooker,a_ME_HB,sizeof(a_ME_HB)/sizeof(a_ME_HB[0]),hist_obtag,histtitle_obtag,"HB","(HB)");
+  bookMESub(ibooker,a_ME_HE,sizeof(a_ME_HE)/sizeof(a_ME_HE[0]),hist_obtag,histtitle_obtag,"HE","(HE)");
+  bookMESub(ibooker,a_ME_HF,sizeof(a_ME_HF)/sizeof(a_ME_HF[0]),hist_obtag,histtitle_obtag,"HF","(HF)");
+  bookMESub(ibooker,a_ME_HE_p,sizeof(a_ME_HE_p)/sizeof(a_ME_HE_p[0]),hist_obtag,histtitle_obtag,"HE_p","(HE+)");
+  bookMESub(ibooker,a_ME_HE_m,sizeof(a_ME_HE_m)/sizeof(a_ME_HE_m[0]),hist_obtag,histtitle_obtag,"HE_m","(HE-)");
+  bookMESub(ibooker,a_ME_HEP17,sizeof(a_ME_HEP17)/sizeof(a_ME_HEP17[0]),hist_obtag,histtitle_obtag,"HEP17","(HEP17)");
+  bookMESub(ibooker,a_ME_HEM17,sizeof(a_ME_HEM17)/sizeof(a_ME_HEM17[0]),hist_obtag,histtitle_obtag,"HEM17","(HEM17)");
+  bookMESub(ibooker,a_ME_HEP18,sizeof(a_ME_HEP18)/sizeof(a_ME_HEP18[0]),hist_obtag,histtitle_obtag,"HEP18","(HEP18)");
+
+/*  histname = hist_obtag +"EtaVsPhi"; histtitle = histtitle_obtag + " eta Vs phi ";
+  bookME(ibooker,jetEtaVsPhi_,histname,histtitle, jet_eta_binning_.nbins, jet_eta_binning_.xmin, jet_eta_binning_.xmax, jet_phi_binning_.nbins, jet_phi_binning_.xmin, jet_phi_binning_.xmax);
+  setMETitle(jetEtaVsPhi_,histtitle_obtag + " #eta","#phi");
+  v_ME.push_back(jetEtaVsPhi_);*/
+  
+  // Initialize the GenericTriggerEventFlag
+  if ( num_genTriggerEventFlag_ && num_genTriggerEventFlag_->on() ) num_genTriggerEventFlag_->initRun( iRun, iSetup );
+  if ( den_genTriggerEventFlag_ && den_genTriggerEventFlag_->on() ) den_genTriggerEventFlag_->initRun( iRun, iSetup );
+
+}
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "DataFormats/Math/interface/deltaR.h" // For Delta R
+void JetMonitor::analyze(edm::Event const& iEvent, edm::EventSetup const& iSetup)  {
+  // Filter out events if Trigger Filtering is requested
+  if (den_genTriggerEventFlag_->on() && ! den_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+  edm::Handle<reco::PFJetCollection> pfjetHandle;
+  iEvent.getByToken( pfjetToken_, pfjetHandle );
+
+  edm::Handle<reco::CaloJetCollection> calojetHandle;
+  iEvent.getByToken( calojetToken_, calojetHandle );
+
+  int ls = iEvent.id().luminosityBlock();
+  v_jetpt.clear();
+  v_jeteta.clear();
+  v_jetphi.clear();
+  std::vector<reco::PFJet> pfjets;
```

Comment:
I didn't try using the edm::View <reco::Jet>.
I focused on reducing copy-and-paste. 
Do you want me to use edm::View <reco::Jet>?